File: Harvard-CS50x/week-2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print the grade level
    //  If grade level is 16 or higher, program should output “Grade 16+”.
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    // If grade level is less than 1, program should output “Before Grade 1”.
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    // Return the number of letters in text
    int lettercount = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            lettercount++;
        }
    }
    return lettercount;
}

int count_words(string text)
{
    // Return the number of words in text
    int wordcount = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            wordcount++;
        }
    }
    // add the last word of the input
    return ++wordcount;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int sentcount = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        switch (text[i])
        {
            case '.':
            case '?':
            case '!':
                sentcount++;
                break;
        }
    }
    return sentcount;
}

File: Harvard-CS50x/week-6/readability.py
# Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8
# L is the average number of letters per 100 words in the text
# S is the average number of sentences per 100 words in the text
# If the resulting index number is 16 or higher, your program should output "Grade 16+"
# If the index number is less than 1, your program should output "Before Grade 1".
# round index to nearest whole number

text = input("Text: ")

l = len(text)
letters = 0
words = 0

for i in range(l):
    if text[i].isalpha():
        letters += 1
    if text[i] == " ":
        words += 1

# count the last word
words += 1

sent = text.count("?") + text.count(".") + text.count("!")


L = (letters / words) * 100
S = (sent / words) * 100

index = round(0.0588 * L - 0.296 * S - 15.8)
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")
